### ISSUE TITLE
NameError: uninitialized constant LAME::Encoding::StereoBufferEncoder::Forwardable

### ISSUE THREAD

**Issue by rabidaudio** [rdvdijk/lame#3]
`StereoBufferEncoder` extends [`Forwardable`](http://www.ruby-doc.org/stdlib-2.2.0/libdoc/forwardable/rdoc/Forwardable.html), which I had to `require` before lame in order to use the gem.

Wouldn't it be preferable for `StereoBufferEncoder` (and any others that use it) to do this? I can send a PR, but Ruby isn't my strongest language so I might be totally missing something here.

**Comment by rr-** [comment]
Having the same issue here.

**Comment by rabidaudio** [comment]
Since the gem is a few years old, I thought it might be an issue with newer versions of Ruby or `ffi`, but I had the same issue on Ruby `1.9.3` with `ffi ~> 1.3.1`.

In any case, this works on both old and new versions.

**Comment by rdvdijk** [comment]
Travis says that the pull request broke some tests... Is that due to your change or is Travis acting up?

**Comment by rabidaudio** [comment]
I tried running the tests before making any changes, and I got a bunch of failures before a segfault (even on `1.9.3`). I suspect it's due to newer versions of some of the gems and unrelated to this. Here are the results if you're interested: https://gist.github.com/rabidaudio/aed7293d7b191398cd53

**Comment by rdvdijk** [comment]
You are correct, the issue is with the new `rspec`. I'll look into and fix it. I suspect that even the `Segmentation fault` is due to newer gems.

**Comment by rdvdijk** [comment]
Thanks for the PR. Before I merge it, can you explain when exactly you get the error? The gem has worked fine in the past without that `require`. I did notice that the stdlib documentation shows the `require` in its first example, but only since Ruby 2.1 (and [not in 2.0](http://www.ruby-doc.org/stdlib-2.0/libdoc/forwardable/rdoc/Forwardable.html)).

**Comment by rabidaudio** [comment]
Reproduction (`rvm reinstall` recompiles ruby and removes all gems):

    rvm reinstall 2.0.0
    gem install lame

Then running:

    ruby -e "require 'lame'"

Results in
```
/home/charles/.rvm/gems/ruby-2.0.0-p598/gems/lame-0.0.3/lib/lame/encoding/stereo_buffer_encoder.rb:4:in `<class:StereoBufferEncoder>': uninitialized constant LAME::Encoding::StereoBufferEncoder::Forwardable (NameError)
	from /home/charles/.rvm/gems/ruby-2.0.0-p598/gems/lame-0.0.3/lib/lame/encoding/stereo_buffer_encoder.rb:3:in `<module:Encoding>'
	from /home/charles/.rvm/gems/ruby-2.0.0-p598/gems/lame-0.0.3/lib/lame/encoding/stereo_buffer_encoder.rb:2:in `<module:LAME>'
	from /home/charles/.rvm/gems/ruby-2.0.0-p598/gems/lame-0.0.3/lib/lame/encoding/stereo_buffer_encoder.rb:1:in `<top (required)>'
	from /home/charles/.rvm/rubies/ruby-2.0.0-p598/lib/ruby/site_ruby/2.0.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /home/charles/.rvm/rubies/ruby-2.0.0-p598/lib/ruby/site_ruby/2.0.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /home/charles/.rvm/gems/ruby-2.0.0-p598/gems/lame-0.0.3/lib/lame.rb:9:in `<top (required)>'
	from /home/charles/.rvm/rubies/ruby-2.0.0-p598/lib/ruby/site_ruby/2.0.0/rubygems/core_ext/kernel_require.rb:128:in `require'
	from /home/charles/.rvm/rubies/ruby-2.0.0-p598/lib/ruby/site_ruby/2.0.0/rubygems/core_ext/kernel_require.rb:128:in `rescue in require'
	from /home/charles/.rvm/rubies/ruby-2.0.0-p598/lib/ruby/site_ruby/2.0.0/rubygems/core_ext/kernel_require.rb:39:in `require'
	from -e:1:in `<main>'
```

Meanwhile:

    ruby -e "require 'forwardable'
    require 'lame'"

no error is returned. That's pretty much all I know; I have no idea why it isn't included by default.

**Comment by rdvdijk** [comment]
Did some research, the `require` is certainly needed, indeed!

One minor thing: Could you move the `require "forwardable"` into the `lib/lame.rb` file?

I'll merge the PR tonight and release a new version.

Thanks for pinpointing this bug and supplying the fix!

**Comment by rdvdijk** [comment]
I've released [version 0.0.4](https://rubygems.org/gems/lame/versions/0.0.4) with your fix.
